packages/angular_devkit/build_angular/src/tools/esbuild/angular/jit-resource-transformer.ts

```diff
@@ -143,82 +143,93 @@ function visitComponentMetadata(
   switch (node.name.text) {
     case 'templateUrl':
       // Only analyze string literals
-      if (
```

Comment:
Note: I couldn't figure out where this logic is tested so I didn't add any tests. I can add some if somebody can point me in the right direction.

Reply:
You can add a test similar to https://github.com/angular/angular-cli/blob/734a46e99ff94d5fffd93f223e4b2b7646790f78/packages/angular_devkit/build_angular/src/builders/application/tests/behavior/component-stylesheets_spec.ts or an E2E https://github.com/angular/angular-cli/tree/main/tests/legacy-cli/e2e/tests/build

Reply:
Hmm okay, it looks like I can't write a test there, because the change hasn't landed in a release yet so writing a component with the new properties results in a compilation error.

Reply:
Actually even if this compiled, it wouldn't be testing the behavior we're looking for, because the transformer only runs on JIT code. As far as I can tell, these tests are running AOT compilation.

Reply:
> Actually even if this compiled, it wouldn't be testing the behavior we're looking for, because the transformer only runs on JIT code. As far as I can tell, these tests are running AOT compilation.

You can switch between AOT and JIT by providing the `aot` as an option in https://github.com/angular/angular-cli/blob/734a46e99ff94d5fffd93f223e4b2b7646790f78/packages/angular_devkit/build_angular/src/builders/application/tests/behavior/component-stylesheets_spec.ts#L20

Let's land it for now, and follow up with a unit test in a separate PR once the changes in the FW are released on NPM.